app/models/transfer_factory.rb

```diff
@@ -1,16 +1,22 @@
 class TransferFactory
-  def initialize(current_organization, current_user, offer_id, destination_account_id)
+  def initialize(current_organization, current_user, offer_id, destination_account_id,
+                 destination_organization_id)
     @current_organization = current_organization
     @current_user = current_user
     @offer_id = offer_id
     @destination_account_id = destination_account_id
+    @destination_organization_id = destination_organization_id.to_i
+  end
+
+  def destination_organization
+    Organization.find(@destination_organization_id)
   end
 
   # Returns the offer that is the subject of the transfer
   #
   # @return [Maybe<Offer>]
   def offer
-    current_organization.offers.find_by_id(offer_id)
+    destination_organization.offers.find_by_id(offer_id)
```

Comment:
I can solve the [codeclimate problem ](https://codeclimate.com/github/coopdevs/timeoverflow/pull/639) by changing this line to` Offer.find(offer_id)` and then changing the line of `destination_organization` to `offer.organization` (and then I would not need to add the parameter) but as this line was already implemented in this form, I don't know if it would be ok.